### ISSUE TITLE
UriTemplateBuilder does not reject variables with invalid characters

### ISSUE THREAD

**Issue by hottrout** [damnhandy/Handy-URI-Templates#52]
```
UriTemplate.buildFromTemplate("location")
  .query("invalid-variable")
  .build();
```

Returns a usable `UriTemplate` that will `expand` and `expandPartial` using this variable name as normal

However

`UriTemplate.fromTemplate("location{?invalid-variable}");`

Throws an exception as the variable name does not match https://tools.ietf.org/html/rfc6570#section-2.3

I think here that `query` should probably be throwing a similar exception.

**Comment by damnhandy** [comment]
Nice catch. That is definitely incorrect.

**Comment by damnhandy** [comment]
The fix is now in the `2.1.6-SNAPSHOT` build, which should be available in Maven Central. I plan on releasing `2.1.6` once I figure out a solution to #51

**Comment by damnhandy** [comment]
`2.1.6` has been released with this fix.
